[PyTorch Edge][tracing-based] Unify tracer between internal and external

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64152

As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

Commit:
[PyTorch Edge][tracing-based] Unify tracer between internal and external



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

malfet commented on torch/csrc/jit/mobile/model_tracer/TracerRunner.cpp:
```diff
@@ -0,0 +1,208 @@
+          mobile_module_runner.get_many_functions_bundled_inputs();
+      for (auto& entry : bundled_inputs_mapping) {
+        std::string function_name = entry.first;
+        std::vector<std::vector<at::IValue>> bundled_inputs = entry.second;
```
Nit
```suggestion
        auto bundled_inputs = entry.second;
```

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] Unify tracer between internal and external"


As title, introduce the file `TracerRunner` shared by internal/external tracer and the main function is
```
TracerResult trace_run(const std::string& input_module_path);
```
which basically takes the path to model file and generate the trace result. The main difference between external tracer and internal tracer is
1. the dependency on `<yaml-cpp/yaml.h>`.
2. the output yaml file from internal tracer includes `model_version` and `model_asset`. These are only needed for internal.



Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30632224/)!

Differential Revision: [D30632224](https://our.internmc.facebook.com/intern/diff/D30632224)

[ghstack-poisoned]
